crates/vm2/src/instruction_handlers/context.rs

```diff
@@ -8,64 +8,75 @@ use zksync_vm2_interface::{
 use super::common::boilerplate;
 use crate::{
     addressing_modes::{Arguments, Destination, Register1, Source},
-    decommit::address_into_u256,
     instruction::ExecutionStatus,
     state::State,
-    Instruction, VirtualMachine,
+    Instruction, VirtualMachine, World,
 };
 
-fn context<T: Tracer, W, Op: ContextOp>(
+pub(crate) fn address_into_u256(address: H160) -> U256 {
+    let mut buffer = [0; 32];
+    buffer[12..].copy_from_slice(address.as_bytes());
+    U256::from_big_endian(&buffer)
+}
+
+fn context<T, W, Op>(
     vm: &mut VirtualMachine<T, W>,
     world: &mut W,
     tracer: &mut T,
-) -> ExecutionStatus {
+) -> ExecutionStatus
+where
+    T: Tracer,
+    W: World<T>,
+    Op: ContextOp,
+{
     boilerplate::<Op, _, _>(vm, world, tracer, |vm, args| {
         let result = Op::get(&vm.state);
-        Register1::set(args, &mut vm.state, result)
+        Register1::set(args, &mut vm.state, result);
     })
 }
 
 trait ContextOp: OpcodeType {
-    fn get<T, W>(state: &State<T, W>) -> U256;
+    fn get<T: Tracer, W: World<T>>(state: &State<T, W>) -> U256;
 }
 
 impl ContextOp for This {
-    fn get<T, W>(state: &State<T, W>) -> U256 {
+    fn get<T: Tracer, W: World<T>>(state: &State<T, W>) -> U256 {
         address_into_u256(state.current_frame.address)
     }
 }
 
 impl ContextOp for Caller {
-    fn get<T, W>(state: &State<T, W>) -> U256 {
+    fn get<T: Tracer, W: World<T>>(state: &State<T, W>) -> U256 {
         address_into_u256(state.current_frame.caller)
     }
 }
 
 impl ContextOp for CodeAddress {
-    fn get<T, W>(state: &State<T, W>) -> U256 {
+    fn get<T: Tracer, W: World<T>>(state: &State<T, W>) -> U256 {
         address_into_u256(state.current_frame.code_address)
     }
 }
 
 impl ContextOp for ErgsLeft {
-    fn get<T, W>(state: &State<T, W>) -> U256 {
-        U256([state.current_frame.gas as u64, 0, 0, 0])
+    fn get<T: Tracer, W: World<T>>(state: &State<T, W>) -> U256 {
+        U256([state.current_frame.gas.into(), 0, 0, 0])
     }
 }
 
 impl ContextOp for ContextU128 {
-    fn get<T, W>(state: &State<T, W>) -> U256 {
+    fn get<T: Tracer, W: World<T>>(state: &State<T, W>) -> U256 {
         state.get_context_u128().into()
     }
 }
 
 impl ContextOp for SP {
-    fn get<T, W>(state: &State<T, W>) -> U256 {
+    fn get<T: Tracer, W: World<T>>(state: &State<T, W>) -> U256 {
         state.current_frame.sp.into()
     }
 }
 
-fn context_meta<T: Tracer, W>(
+#[allow(clippy::cast_sign_loss)] // intentional
```

Comment:
I believe you can put this annotation before the offending line to make it more obvious.